mesheryctl/internal/cli/root/system/reset.go

```diff
@@ -58,16 +62,73 @@ func resetMesheryConfig() error {
 	}
 
 	log.Info("Meshery resetting...\n")
-	log.Printf("Fetching default docker-compose file as per current-context: %s...\n", mctlCfg.CurrentContext)
-	err = utils.DownloadDockerComposeFile(currCtx, true)
-	if err != nil {
-		return errors.Wrap(err, "failed to fetch docker-compose file")
-	}
-
 	log.Printf("Current Context: %s", mctlCfg.CurrentContext)
 	log.Printf("Channel: %s", currCtx.Channel)
-	log.Printf("Version: %s\n", currCtx.Version)
+	log.Printf("Version: %s", currCtx.Version)
+	log.Printf("Platform: %s\n", currCtx.Platform)
+
+	switch currCtx.Platform {
+	case "docker":
+
+		log.Printf("Fetching default docker-compose file as per current-context: %s...\n", mctlCfg.CurrentContext)
+		err = utils.DownloadDockerComposeFile(currCtx, true)
+		if err != nil {
+			return errors.Wrap(err, "failed to fetch docker-compose file")
+		}
+		log.Info("...Meshery config (" + utils.DockerComposeFile + ") now reset to default settings.")
+
+	case "kubernetes":
 
-	log.Info("...Meshery config (" + utils.DockerComposeFile + ") now reset to default settings.")
+		version := currCtx.Version
+		if version == "latest" {
+			if currCtx.Channel == "edge" {
+				version = "master"
+			} else {
+				version, err = utils.GetLatestStableReleaseTag()
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		log.Debug("fetching required Kubernetes manifest files...")
+		// get correct minfestsURL based on version
+		manifestsURL, err := utils.GetManifestTreeURL(version)
```

Comment:
Could we add this to the start command too? I think by current logic, it picks up only the latest version.

Reply:
Yes, the start command picks up only the latest version.
Should I update this PR or do it in a new one?

Reply:
You can update this PR to add that and mention it in the description.

Reply:
In part, the question here becomes that of centralizing this functionality.

The helper script(s) have become too bloated and need breaking out into different files.

Reply:
@leecalcote I agree. There are many platform/ component specific functions that are in the helpers.go file and it is difficult to maintain.